### ISSUE TITLE
linter: allow opting out of the hardcoded "minified" filename skip

### ISSUE THREAD

**Issue by levithomason** [oxc-project/oxc#26163]
> Disclosure per the [AI usage policy](https://github.com/oxc-project/oxc/blob/main/CONTRIBUTING.md#ai-usage-policy): this issue was drafted with AI assistance. Every command, output, and code reference below was run/verified by me before posting, and I'm responsible for its content.

### What version of Oxlint are you using?

1.80.0 (filter is identical on `main`)

### What command did you run?

`oxlint --no-ignore src` — but any invocation, including passing the file explicitly.

### Summary

`Walk::is_wanted_entry` in [`apps/oxlint/src/walk.rs`](https://github.com/oxc-project/oxc/blob/main/apps/oxlint/src/walk.rs#L140-L142) drops any file whose **basename** contains `.min.`, `-min.` or `_min.`:

```rust
if [".min.", "-min.", "_min."].iter().any(|e| file_name.contains(e)) {
    return false;
}
```

The heuristic is right for `jquery.min.js`. It is wrong for a source file where `min` means **minimum**, and it is unconditional: it sits outside the `if !options.no_ignore { … }` block, so no ignore layer, config key, CLI flag or env var reaches it. I'd like an opt-out.

### What happened

A source file named `criteria-allocation-balance-min.ts` (`min` = minimum) in our trading system was silently never linted. It's an entry criterion for live orders, and it lost `no-floating-promises`, `await-thenable` and `no-misused-promises` coverage without any signal — I only found it because we run a separate guard that asserts every file reaches the type checker.

It's a pair, not a single file: the colocated test file's basename matches too, so working around it means renaming both. With our rename reverted, the guard reports two gaps:

```
✗ type-aware lint covered less than the repo (2 gaps):
  oxlint's walker skipped this name as a minified bundle (.min. / -min. / _min.), so it was not checked — rename the file: packages/core/src/criteria/criteria-allocation-balance-min.ts
  oxlint's walker skipped this name as a minified bundle (.min. / -min. / _min.), so it was not checked — rename the file: packages/core/tests/criteria/criteria-allocation-balance-min.test.ts
```

Minimal reproduction:

```sh
mkdir -p src
for n in a-min a.min a_min a-min-b a-plain min; do
  printf 'export const x = 1;\ndebugger;\n' > "src/$n.ts"
done
oxlint --no-ignore src
```

```
src/min.ts:2:1: warning eslint(no-debugger): `debugger` statement is not allowed
src/a-plain.ts:2:1: warning eslint(no-debugger): `debugger` statement is not allowed
src/a-min-b.ts:2:1: warning eslint(no-debugger): `debugger` statement is not allowed
```

`a-min.ts`, `a.min.ts` and `a_min.ts` are missing. The shape of the trap, verified on 1.80.0:

| filename | linted? |
| --- | --- |
| `a-min.ts` / `a.min.ts` / `a_min.ts` | **no** |
| `a-min.test.ts` / `a-min.d.ts` | **no** |
| `a-min-b.ts` | yes |
| `min.ts` | yes |
| `foo-min/bar.ts` | yes (directories aren't matched) |

**The loss is not observable from oxlint's own output.** On a clean tree, `oxlint --no-ignore clean/` (containing only `a-min.ts`) prints nothing and exits 0 — no diagnostic, no summary, no non-zero status. Explicit paths don't help either; they go through the same walker:

```
$ oxlint src/a-min.ts src/a-plain.ts
src/a-plain.ts:2:1: warning eslint(no-debugger): ...
```

Only `a-plain.ts` is reported, and nothing says `a-min.ts` was dropped. (Passing the skipped file *alone* is the one case that surfaces: `No files found to lint.`) A project cannot detect this from oxlint; it has to build its own coverage check outside the tool.

For reference, `oxfmt` has no such filter — [`apps/oxfmt/src/cli/walk.rs`](https://github.com/oxc-project/oxc/blob/main/apps/oxfmt/src/cli/walk.rs) skips only VCS dirs and `node_modules` — so `*-min.ts` is formatted but not linted.

### Proposal

Make the skip opt-out-able, exactly as `oxfmt` already does for `node_modules`. From [`apps/oxfmt/src/cli/command.rs`](https://github.com/oxc-project/oxc/blob/main/apps/oxfmt/src/cli/command.rs#L155-L164):

```rust
/// Format code in node_modules directory (skipped by default)
#[bpaf(switch, hide_usage)]
pub with_node_modules: bool,
```

The oxlint equivalent would be a `with_minified: bool` on [`apps/oxlint/src/command/ignore.rs`](https://github.com/oxc-project/oxc/blob/main/apps/oxlint/src/command/ignore.rs)'s `IgnoreOptions`, threaded into `Walk` alongside `extensions` (`Walk::new` already takes `&IgnoreOptions`) and read in `is_wanted_entry`. Default unchanged; `--with-minified` opts in. A mirroring `options.withMinified` config key would follow the existing convention that every `options` member is "equivalent to passing `--x` on the CLI", if you want it.

This is the shape @leaysgur described in #25259 for a different hardcoded skip — *"add a hardcoded skip + opt-out later if requested (like `oxfmt --with-node-modules`)"* — so I'm requesting it.

Origin of the filter is 7caaa0490 (2023-03-11, "feat(cli): ignore files with `min` suffix"), with no stated rationale, and I couldn't find an existing issue for it.

### The PR is ready

Implemented and pushed at https://github.com/levithomason/oxc/tree/feat/oxlint-opt-out-of-minified-skip; a PR follows immediately.

It adds `with_minified: bool` to `IgnoreOptions` (`#[bpaf(switch, hide_usage)]`, giving `--with-minified`) and threads it into `Walk` alongside `extensions`, exactly as `oxfmt` threads `with_node_modules`. The existing filter is guarded with `!with_minified`, so default behavior is byte-identical. Adds a `walk.rs` test asserting both halves of the contract and a CLI-parsing test; the two `tasks/website_linter` CLI snapshots are regenerated for the new flag.

I left `options.withMinified` out to keep the PR minimal — happy to add it in the same PR if you want the config key to mirror the flag.
